partner_communication_switzerland/models/res_partner.py

```diff
@@ -179,41 +179,48 @@ def _compute_no_physical_letter(self):
 
     def _inverse_no_physical_letter(self):
         for partner in self:
-            if partner.no_physical_letter:
-                vals = {
-                    "nbmag": "no_mag" if partner.nbmag == "no_mag" else "email",
-                    "tax_certificate": "no"
-                    if partner.tax_certificate == "no" else "only_email",
-                    "calendar": False,
-                    "christmas_card": False
-                }
-                for _field in ["global_communication_delivery_preference",
-                               "letter_delivery_preference",
-                               "photo_delivery_preference",
-                               "thankyou_preference"]:
-                    value = getattr(partner, _field)
-                    if "auto" in value or value == "both":
-                        vals[_field] = "auto_digital_only"
-                    elif value in ["physical", "digital"]:
-                        vals[_field] = "digital_only"
-                partner.write(vals)
-            else:
-                vals = {
-                    "calendar": True,
-                    "christmas_card": True
-                }
-                for _field in ["global_communication_delivery_preference",
-                               "letter_delivery_preference",
-                               "photo_delivery_preference",
-                               "thankyou_preference"]:
-                    value = getattr(partner, _field)
-                    if "only" in value:
-                        vals[_field] = value.replace("_only", "")
-                if partner.nbmag == "no_mag":
-                    vals["nbmag"] = "one"
-                if partner.tax_certificate == "only_email":
-                    vals["tax_certificate"] = "default"
-                partner.write(vals)
+            self.env.with_context(no_physical_letters=partner.no_physical_letter)
+            partner.compute_inverse_no_physical_letter()
+
+    def compute_inverse_no_physical_letter(self):
+        self.ensure_one()
+        no_physical_letters = self.env.context['no_physical_letters']
```

Comment:
I would do it like this:
```suggestion
            partner.compute_inverse_no_physical_letter()

    def compute_inverse_no_physical_letter(self):
        self.ensure_one()
        no_physical_letters = self.env.context.get("no_physical_letters", partner.no_physical_letter)
```